src/main/java/com/yelp/nrtsearch/server/luceneserver/field/properties/TermQueryable.java

```diff
@@ -25,18 +26,130 @@
  */
 public interface TermQueryable {
   /**
-   * Build a term query for this field type with the given configuration.
+   * Performs type validation. Do not @Override in subclasses.
+   *
+   * <p>Build a term query for this field type with the given configuration.
    *
    * @param termQuery term query configuration
    * @return lucene term query
    */
-  Query getTermQuery(TermQuery termQuery);
+  default Query getTermQuery(TermQuery termQuery) {
+    Query query;
+    switch (termQuery.getTermTypesCase()) {
+      case BOOLEANVALUE:
+        query = getTermQueryFromBooleanValue(termQuery.getBooleanValue());
+        break;
+      case DOUBLEVALUE:
+        query = getTermQueryFromDoubleValue(termQuery.getDoubleValue());
+        break;
+      case FLOATVALUE:
+        query = getTermQueryFromFloatValue(termQuery.getFloatValue());
+        break;
+      case INTVALUE:
+        query = getTermQueryFromIntValue(termQuery.getIntValue());
+        break;
+      case LONGVALUE:
+        query = getTermQueryFromLongValue(termQuery.getLongValue());
+        break;
+      case TEXTVALUE:
+        query = getTermQueryFromTextValue(termQuery.getTextValue());
+        break;
+      default:
+        query = null;
+        break;
+    }
+
+    if (query.equals(null)) {
```

Comment:
This will throw a NullPointerException if query is null. You will want to use reference equality, i.e. `query == null`